### PR TITLE
DofTools: add missing codimension-one instantiation

### PR DESCRIPTION
**Pull request by tamiko**
In reference to #16493

### CODE REVIEW

**Review by peterrum — APPROVED** [review]

File: source/dofs/dof_tools_sparsity.inst.in [hunk 1]
```diff
@@ -257,15 +257,17 @@ for (deal_II_dimension : DIMENSIONS)
   }
 
 
-for (deal_II_dimension : DIMENSIONS)
+for (deal_II_dimension, deal_II_space_dimension : DIMENSIONS)
```
Comment (marcfehling) [inline source/dofs/dof_tools_sparsity.inst.in:260]:
```suggestion
for (deal_II_dimension : DIMENSIONS; deal_II_space_dimension : SPACE_DIMENSIONS)
```
We have split dim and spacedim into two separate template argument lists in other occassions.

See also
https://github.com/dealii/dealii/blob/20d2176d317137ec1b5c8fc1334cfef649765c14/cmake/config/template-arguments.in#L254
https://github.com/dealii/dealii/blob/a65d7b464723ebdb141d1f648074e9a844342dc6/source/grid/tria.inst.in#L18

Comment (marcfehling) [reply source/dofs/dof_tools_sparsity.inst.in:260]: Hmm, we wrote `(deal_II_dimension, deal_II_space_dimension : DIMENSIONS)` also in other parts of the library. Maybe it's better to discuss it in a separate PR. I will prepare one.

**Review by marcfehling — APPROVED** [review]

File: source/dofs/dof_tools_sparsity.inst.in (hunk 1 above)
Comment (tamiko) [reply source/dofs/dof_tools_sparsity.inst.in:260]: @marcfehling Yes, let's fix that up in a separate PR so that we catch all occurences.

**Pull request merged**